upa/contracts/UpaVerifier.sol

```diff
@@ -377,26 +391,35 @@ contract UpaVerifier is
         verifiedSubmissionHeight = submissionBlockNumber;
     }
 
-    function verifyAggregatedSubmissionProof(
+    /// Verify an aggregated proof with the `sidOuterVerifier`.
+    function verifyAggregatedProofSid(
         bytes calldata proof
     ) external onlyWorker {
-        // Extract submission Id
+        // Compute submissionId from the calldata
         uint256 proofL;
         uint256 proofH;
         assembly {
             proofL := calldataload(add(proof.offset, /* 12 * 0x20 */ 0x180))
             proofH := calldataload(add(proof.offset, /* 13 * 0x20 */ 0x1a0))
         }
-        bytes32 submissionId = UpaLib.digestFromFieldElements(proofL, proofH);
+        bytes32 submissionId = UpaLib.fieldElementsAsDigest(proofL, proofH);
 
-        // Verify proof
-        (bool success, ) = _getVerifierStorage().outerVerifier.call(proof);
+        VerifierStorage storage verifierStorage = _getVerifierStorage();
+
+        // require there is a sidOuterVerifier contract
+        require(
```

Comment:
Seems like something we should assert up front.
Also, I don't see the equivalent check for the `verifyAggregatedProof` case.

Reply:
Found the other check.

Reply:
Ah, I mistook the location.  Please ignore.